source/server/admin/config_dump_handler.cc

```diff
@@ -129,11 +143,17 @@ Http::Code ConfigDumpHandler::handlerConfigDump(absl::string_view url,
   if (resource.has_value()) {
     auto err = addResourceToDump(dump, mask, resource.value(), include_eds);
     if (err.has_value()) {
+      response_headers.setReferenceContentType(Http::Headers::get().ContentTypeValues.Text);
       response.add(err.value().second);
       return err.value().first;
     }
   } else {
-    addAllConfigToDump(dump, mask, include_eds);
+    auto err = addAllConfigToDump(dump, mask, include_eds);
+    if (err.has_value()) {
+      response_headers.setReferenceContentType(Http::Headers::get().ContentTypeValues.Text);
```

Comment:
sorry can you also add
```
   response_headers.addReference(
       Http::Headers::get().XContentTypeOptions,
       Http::Headers::get().XContentTypeOptionValues.Nosniff);
```
Just a little paranoia to make sure browsers don't flip into HTML mode if some text looks like HTML. Just a bit of paranoia based on past browser behavior :)

Reply:
Done.